pkg/validation/attributes/attributes_event.go

```diff
@@ -0,0 +1,45 @@
+package attributes
+
+import (
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
+)
+
+// ValidateEvents validates the events data for a global attribute
+func ValidateEvents(attributes apiAttributes.Attributes, events *v1alpha2.Events) error {
+	var err error
+
+	if events != nil {
+		switch {
+		case len(events.PreStart) > 0:
+			for i := range events.PreStart {
+				if events.PreStart[i], err = validateAndReplaceDataWithAttribute(events.PreStart[i], attributes); err != nil {
```

Comment:
I'm curious on the design here -- `events.Prestart` is a slice of strings referring to command IDs, and command IDs cannot have `{` and `}` in them, which makes it seem impossible to cleanly use this functionality. I'm not sure we should support attributes inside `events` unless we relax the regexes for validation (though that poses further problems)

Reply:
I think this is a valid point 👍

Reply:
Addressed